contracts/FractalModule.sol

```diff
@@ -0,0 +1,95 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@gnosis.pm/zodiac/contracts/core/Module.sol";
+import "./interfaces/IFractalModule.sol";
+
+contract FractalModule is IFractalModule, Module {
+    mapping(address => bool) public controllers; // A DAO may authorize users to act on the behalf of the parent DAO.
+
+    /// @dev Throws if called by any account other than the owner.
+    modifier onlyAuthorized() {
+        require(
+            owner() == msg.sender || controllers[msg.sender],
+            "Not Authorized"
+        );
+        _;
+    }
+
+    /// @dev Initialize function
+    /// @param initializeParams Parameters of initialization encoded
+    function setUp(bytes memory initializeParams) public override initializer {
+        __Ownable_init();
+        (
+            address _owner, // Controlling DAO
+            address _avatar, // GSafe // Address(0) == msg.sender
+            address _target, // GSafe or Modifier  // Address(0) == msg.sender
+            address[] memory _controllers // Authorized controllers
+        ) = abi.decode(
+                initializeParams,
+                (address, address, address, address[])
+            );
+
+        setAvatar(_avatar == address(0) ? msg.sender : _avatar);
+        setTarget(_target == address(0) ? msg.sender : _target);
+        addControllers(_controllers);
+        transferOwnership(_owner);
+    }
+
+    /// @notice Allows an authorized user to exec a Gnosis Safe tx via the module
+    /// @param execTxData Data payload of module transaction.
+    function batchExecTxs(bytes memory execTxData) public onlyAuthorized {
```

Comment:
We should probably change the name of this function now that it only executes one transaction